service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -79,9 +80,20 @@ public OrderTypeAndDocument buildOrderTypeAndDocument(OrderTypeAndDocument typeA
             case BLANK_ORDER:
                 orderBuilder.title(defaultIfBlank(generatedOrder.getTitle(), "Order"));
                 orderBuilder.details(generatedOrder.getDetails());
+                orderBuilder.expiryDate(null);
                 break;
             case CARE_ORDER:
                 orderBuilder.title(null);
+                orderBuilder.expiryDate(null);
+                break;
+            case SUPERVISION_ORDER:
+                orderBuilder.title(null);
+                String date = null;
+                if (orderMonths != null) {
+                    date = dateFormatterService.formatLocalDateTimeBaseUsingFormat(time.now().plusMonths(orderMonths),
+                        "h:mma, d MMMM y");
+                }
+                orderBuilder.expiryDate(date);
```

Comment:
minor: can be:
                ofNullable(orderMonths)
                    .map(m-> time.now().plusMonths(m))
                    .map(d-> dateFormatterService.formatLocalDateTimeBaseUsingFormat(d, "h:mma, d MMMM y"))
                    .ifPresent(orderBuilder::expiryDate);